source/aspnetintegration.rst

```diff
@@ -263,3 +266,210 @@ A typical solution to this problem is to split up the class into multiple smalle
 Simple Injector :ref:`promotes <Push-developers-into-best-practices>` best practices, and because of downsides described above, we consider the use of the `[FromServices]` attribute *not* to be a best practice. This is why we choose not to provide out-of-the-box support for injecting Simple Injector registered dependencies into controller actions. 
 
 In case you still feel method injection is the best option for you, you can plug in a custom `IModelBinderProvider` implementation returning a custom `IModelBinder` that resolves instances from Simple Injector.
+
+.. _using-simpleinjector-with-aspnetcore-ihostedservices:
+
+Using Simple Injector with ASP.NET Core 2.0 IHostedServices
+===========================================================
+
+There are multiple ways to integrate Simple Injector with IHostedServices.  The easiest approach is to cross-wire the hosted service with the ASP.NET core configuration system so that it resolves the hosted service implementation from Simple Injector.  That would look something like this:
+
+.. code-block:: c#
```

Comment:
Can we minimize this example to something like this?:

``` c#
// Cross-wire MyHostedService with the ASP.NET Core configuration system
services.AddSingleton<IHostedService>(c => container.GetInstance<MyHostedService>());

// Register MyHostedService in Simple Injector as Singleton
container.RegisterSingleton<MyHostedService>();
```

My  reasoning is as follows:

* The display of the `MyHostedService` example feels to verbose. A reader will likely have read the ASP.NET Core documentation on defining hosted services (although we might through in a link to that documentation).
* The `MyHostedService` depends on an `ISettingsObject`, which function is unclear.
* The `Startup` class duplicates much of the registration of the page's quick-start example. This might make the user loose track of the actual essential changes, which is actually 2 lines of code.
* The comment about the delegate being called after the container has been configured is not that important for most readers. It was essential for you in understanding _why_ this works, but most readers will simply copy&paste the example and assume it works (as it should, as we present it in the documentation). For them, such comment will only be noise.
* By leaving out the `ConfigureServices` method signature, it allows leaving out the usual code that's inside that method. Otherwise, when the example would start with `public void ConfigureServices(IServiceCollection service) { ...`, it would become confusing if all the usual stuff (like configuring MVC and Simple Injector) would be removed. By simply showing the code to add, we remove that confusion.  *Note that it would still be useful to describe in the text that this code should be added to the `IntegrateSimpleInjector` method*. 

Does that make sense?

Reply:
I see what you're saying.  I included the implementation of the MyHostedService because it was in your answer on StackOverflow and it was really helpful for me when I first implemented it myself.  When I read and write documentation for some of the other code that I work on, I tend to be more verbose in style.  But I can see how all of the detail might not be absolutely necessary and might be distracting for people.  

How about if I reduced it down to something like this:

    //See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services for full implementation
    //details.
    //The constructor of the hosted service can be written like all other application-specific components and
    //contain dependencies which the Simple Injector container will resolve when it resolves this instance.
    //For example, if a hosted service took application-specific dependencies of ISettingsObject and ILogger:
    public MyHostedService(ISettingsObject settings, ILogger logger)
    {
        _settingsObj = settings;
        _logger = logger;
    }

The hosted service can then be cross-wired with the ASP.NET configuration system in the `IntegrateSimpleInjector` method:

    services.AddSingleton<IHostedService>(c => container.GetInstance<MyHostedService>());

Finally, register the hosted service as a Singleton in the `InitializeContainer` method.  Hosted Services are implemented internally by ASP.NET Core as Singletons.

    container.RegisterSingleton<MyHostedService>();

Reply:
Please note that code listings should not exceed 90 characters in width. Wider listings won't be displayed correctly in the [PDF](https://readthedocs.org/projects/simpleinjector/downloads/pdf/latest/) generated by readthedocs.

Reply:
I'd still prefer removing `ISettingsObject` to simplify the example.